libs/navigation/api/src/lib/navigation.resolver.ts

```diff
@@ -0,0 +1,58 @@
+import {Args, Mutation, Query, Resolver} from '@nestjs/graphql'
+import {
+  CanCreateNavigation,
+  CanDeleteNavigation,
+  CanGetNavigations,
+  Permissions
+} from '@wepublish/permissions/api'
+import {
+  CreateNavigationInput,
+  Navigation,
+  NavigationArgs,
+  NavigationIdArgs,
+  UpdateNavigationArgs
+} from './navigation.model'
+import {NavigationService} from './navigation.service'
+import {BadRequestException} from '@nestjs/common'
+
+@Resolver(() => Navigation)
+export class NavigationResolver {
+  constructor(private readonly navigationService: NavigationService) {}
+
+  @Query(() => Navigation, {nullable: true})
+  getNavigation(@Args() {id, key}: NavigationArgs) {
+    if (id) {
+      return this.navigationService.getNavigationById(id)
+    }
+
+        if (key) {
+      return this.navigationService.getNavigationByKey(key)
+    }
+
+    throw new BadRequestException('Need to provide id or key')
+  }
+
+  @Query(() => [Navigation])
+  @Permissions(CanGetNavigations)
+  getNavigations() {
+    return this.navigationService.getNavigations()
+  }
+
+  @Mutation(() => Navigation)
+  @Permissions(CanCreateNavigation)
+  createNavigation(@Args('input') input: CreateNavigationInput) {
```

Comment:
Don't use input as an argument please, there's no need to have  `{input: {...}}` and not `{...}`